go/api/commands.go

```diff
@@ -731,4 +731,96 @@ type ConnectionManagementCommands interface {
 	//
 	// [valkey.io]: https://valkey.io/commands/ping/
 	PingWithMessage(message string) (string, error)
+
+	// ReplaceConnectionPassword updates the connection password and optionally re-authenticates.
+	//
+	// See [valkey.io] for details.
+	//
+	// Parameters:
+	//  password - The new password to set.
+	//  reAuth - If true, re-authenticates the connection with the new password.
+	//
+	// Return value:
+	//  A Result containing the operation status, or an error if the password replacement fails.
+	//
+	// For example:
+	//  result, err := client.ReplaceConnectionPassword("newpass123", true)
+	//  // result equals api.CreateStringResult("OK")
+	//
+	// [valkey.io]: https://valkey.io/commands/auth/
+	ReplaceConnectionPassword(password string, reAuth bool) (Result[string], error)
+}
+
+// ServerManagementClusterCommands provides cluster management operations.
+//
+// See [valkey.io] for details.
+//
+// [valkey.io]: https://valkey.io/commands/?group=server
+type ServerManagementClusterCommands interface {
```

Comment:
ServerManagementClusterCommands need to have a different definition of `config get`, `config set` and `custom command`. Please refer the implementation of other clients.

Reply:
@janhavigupta007 It's done.